language/libra-vm/src/libra_transaction_executor.rs

```diff
@@ -307,23 +307,26 @@ impl LibraVM {
         &self,
         remote_cache: &StateViewCache<'_>,
         writeset_payload: &WriteSetPayload,
+        txn_data: Option<&TransactionMetadata>,
     ) -> Result<ChangeSet, Result<(VMStatus, TransactionOutput), VMStatus>> {
         let gas_schedule = zero_cost_schedule();
         let mut cost_strategy = CostStrategy::system(&gas_schedule, GasUnits::new(0));
 
         Ok(match writeset_payload {
             WriteSetPayload::Direct(change_set) => change_set.clone(),
-            WriteSetPayload::Script {
-                script,
-                execute_as: signer,
-            } => {
+            WriteSetPayload::Script { script, execute_as } => {
                 let mut tmp_session = self.0.new_session(remote_cache);
+                let mut args = convert_txn_args(script.args());
+                args.insert(0, Value::transaction_argument_signer_reference(*execute_as));
```

Comment:
@tnowacki This is where the other signer is added. It then follows the normal transaction flow that will add the sender's `&signer` to the arguments.

Reply:
I don't think we should be doing this at this point if we can make do with what we have. 

Changing the rules for the bytecode verifier and how we feed in signer information into an admin script is a pretty major change. Especially since the bytecode verifier now accepts all sorts of different scripts it didn't before (granted there is no way to actually invoke them, but it is still a large change)

I'll book some time to discuss this